src/cpp/session/SessionMain.cpp

```diff
@@ -1701,7 +1702,7 @@ int saveWorkspaceAction()
 
 void syncRSaveAction()
 {
-   rstudio::r::session::setSaveAction(saveWorkspaceOption());
```

Comment:
`saveWorkspaceOption()` is the variant that returns an R `SA_TYPE`. Unfortunately, since `enum` and `int` are compatible types, the C++ type system let us shoot ourselves in the foot here 😞